bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java

```diff
@@ -51,80 +66,189 @@
+
+            String id = getThing().getUID().getId();
+            gardenaSmart = new GardenaSmartImpl(id, gardenaConfig, this, scheduler, httpClientFactory,
+                    webSocketFactory);
+            final GardenaDeviceDiscoveryService discoveryService = this.discoveryService;
+            if (discoveryService != null) {
+                discoveryService.startScan(null);
+                discoveryService.waitForScanFinishing();
+            }
+            apiCallSuppressionUpdate(Duration.ZERO);
+            updateStatus(ThingStatus.ONLINE);
+        } catch (GardenaException ex) {
+            logger.warn("{}", ex.getMessage());
+            synchronized (reInitializationCodeLock) {
+                Duration delay;
+                int status = ex.getStatus();
+                if (status <= 0) {
+                    delay = REINITIALIZE_DELAY_SECONDS;
+                } else if (status == HttpStatus.TOO_MANY_REQUESTS_429) {
+                    delay = REINITIALIZE_DELAY_HOURS_LIMIT_EXCEEDED;
                 } else {
-                    // otherwise reinitialize after 120 seconds
-                    scheduleReinitialize(REINITIALIZE_DELAY_SECONDS, TimeUnit.SECONDS);
+                    delay = apiCallSuppressionDelay().plus(REINITIALIZE_DELAY_MINUTES_BACK_OFF);
                 }
-                logger.warn("{}", ex.getMessage());
+                scheduleReinitialize(delay);
+                apiCallSuppressionUpdate(delay);
             }
-        });
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, getUiText());
+            disposeGardena();
+        }
+    }
+
+    /**
+     * Re-initializes the GardenaSmart account.
+     * This method is called on a background thread.
+     */
+    private synchronized void reIninitializeGardena() {
+        if (getThing().getStatus() != ThingStatus.UNINITIALIZED) {
+            initializeGardena();
+        }
     }
 
     /**
      * Schedules a reinitialization, if Gardena smart system account is not reachable.
      */
-    private void scheduleReinitialize(long delay, TimeUnit unit) {
-        scheduler.schedule(() -> {
-            if (getThing().getStatus() != ThingStatus.UNINITIALIZED) {
-                initializeGardena();
-            }
-        }, delay, unit);
+    private void scheduleReinitialize(Duration delay) {
+        ScheduledFuture<?> reInitializationTask = this.reInitializationTask;
+        if (reInitializationTask != null) {
+            reInitializationTask.cancel(false);
+        }
+        this.reInitializationTask = scheduler.schedule(() -> reIninitializeGardena(), delay.getSeconds(),
+                TimeUnit.SECONDS);
     }
 
     @Override
     public void dispose() {
         super.dispose();
+        synchronized (reInitializationCodeLock) {
+            ScheduledFuture<?> reInitializeTask = this.reInitializationTask;
+            if (reInitializeTask != null) {
+                reInitializeTask.cancel(false);
```

Comment:
Since this is dispose, I suppose we need to interrupt the task?
```suggestion
                reInitializeTask.cancel(true);
```

Reply:
Done.